cartridge-gg/controller
# 🏆  Make connection between world and trophies

This PR add the communication between the indexer data from the game's world and the controller.
Notes:
- It adds a namespace option to the controller, required to fetch the correct data from the world
- It adds FontAwesome icon files directly in the html (FontAwesome guide) to be able to display any solid icon dynamically (note that it doesn't work very well in dev mode with re-rendering). The files are only 2Mb big so I guess it's fine and I didnt added any lazy imports.
- The data are fetched and processed inside hooks, it works well with few data, I could imagine that it will bring some latency issue (there is already a loader page) if there is a bunch of data to fetch so it should probably be migrated server side (and mb with a DB or any cache system). With the v1-rc0 or dojo, all this fetching work will probably change since we will probably leave the raw events and use directly models since they won't be overridden anymore when a new event with the same keys is emitted.
- I have added a `models` folder with few classes to have good places for types and parsing related to types to avoid having a big file with many parsers for a bit everything

To test it, you can:
- Update your .env.development with the following:
```txt
NEXT_PUBLIC_INDEXER_URL="https://api.cartridge.gg/x/conquest/torii/graphql"
NEXT_PUBLIC_NAMESPACE="conquest"
```

- Connect and click on some buttons in here:

  - https://conquest-one.vercel.app/

Note: except if you succeed to use your prod controller inside the Profile example page, you can see your activity within the leaderboard tab of the Trophy profile page.
